### PR TITLE
Refactor CircleCI config

### PR DESCRIPTION
**Pull request by dtrudg**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,7 +6,30 @@ executors:
...
     docker:
-      - image: python:3.9-buster
+      - image: python:3.9-bullseye
+
+commands:
+  install-latex-apt:
+    parameters:
+        sudo:
+          type: boolean
+          default: false
```
Comment (tri-adam) [inline .circleci/config.yml:16]: Can probably drop this, I can't see any jobs that set it?

File: .circleci/config.yml [hunk 2]
```diff
@@ -6,7 +6,30 @@ executors:
...
+          name: Update package indexes
+          command: <<# parameters.sudo >>sudo <</ parameters.sudo >>apt-get -q update
+      - run:
+          name: Install LaTeX environment
+          command: <<# parameters.sudo >>sudo <</ parameters.sudo >>apt-get -q install -y texlive-latex-extra latexmk
+  install-deps-python:
+    parameters:
+      sudo:
+        type: boolean
+        default: false
```
Comment (tri-adam) [inline .circleci/config.yml:28]: Same here?

**Review by tri-adam — COMMENTED** [review]
Recommend removing unused `parameters` to simplify the config.

File: .circleci/config.yml (hunk 1 above)
Comment (dtrudg) [reply .circleci/config.yml:16]: Yeah - I left it in from the other repos where we have mixed docker and machine environments... that's not the case here (at present)... I can always add it back if needed.

**Review by tri-adam — APPROVED** [review]

**Pull request merged**